erdjs/src/smartcontracts/typesystem/tokenIdentifier.ts

```diff
@@ -0,0 +1,32 @@
+import * as errors from "../../errors";
+import { PrimitiveType, PrimitiveValue } from "./types";
+
+export class TokenIdentifierType extends PrimitiveType {
+    constructor() {
+        super("TokenIdentifier");
+    }
+}
+
+export class TokenIdentifierValue extends PrimitiveValue {
+    private readonly value: Buffer;
+
+    constructor(value: Buffer) {
+        super(new TokenIdentifierType());
+        this.value = value;
+    }
+
+    getLength(): number {
+        return this.value.length;
+    }
+
+    /**
+     * Returns whether two objects have the same value.
+     */
+    equals(other: TokenIdentifierValue): boolean {
+        return this.value.equals(other.value);
```

Comment:
Maybe compare lengths first.

Reply:
Fixed (though I hope this is also implemented by `buffer.equals()` - as far as I know, this delegates to `buffer.compare()`, but I didn't check the source code (nodeJS / browsers).